src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -3,13 +3,17 @@
 import java.time.LocalDate;
 import java.util.List;
 import java.util.Optional;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 import wolox.training.models.User;
 
 public interface UserRepository extends CrudRepository<User, Long> {
 
 	Optional<User> findByUsername(String username);
 
-	List<User> findByBirthdateBetweenAndNameContainingIgnoreCase(LocalDate startDate, LocalDate endDate,String name);
+	@Query("SELECT u FROM User u WHERE (u.birthdate BETWEEN :startDate  AND :endDate) and (lower(u.name) like lower(concat('%', :name,'%')))")
```

Comment:
If the birthdate is NULL how would handle this query that case? Also for the name parameter, you are not handling the case when it is NULL.

Also, I think it would be better to do concat('%', lower(:name),'%') instead of using lower in the full concatenated string.